编译openldap时出错

如题：
```
openldap-2.6.7/libraries/libldap/tls_o.c:1075:(.text+0x5a1): undefined reference to `EVP_md2'
bin/ld: openldap-2.6.7/libraries/libldap/tls_o.c:1078:(.text+0x5af): undefined reference to `EVP_md4'
bin/ld: ./.libs/libldap.a(tls_o.o): in function `tlso_session_endpoint':
openldap-2.6.7/libraries/libldap/tls_o.c:1075:(.text+0x5a1): undefined reference to `EVP_md2'
bin/ld: openldap-2.6.7/libraries/libldap/tls_o.c:1078:(.text+0x5af): undefined reference to `EVP_md4'
./.libs/libldap.a(tls_o.o): in function `tlso_session_endpoint':
openldap-2.6.7/libraries/libldap/tls_o.c:1075:(.text+0x5a1): undefined reference to `EVP_md2'
ld: /openldap-2.6.7/libraries/libldap/tls_o.c:1078:(.text+0x5af): undefined reference to `EVP_md4'
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:336: urltest] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: *** [Makefile:324: apitest] Error 1
make[4]: *** [Makefile:328: ftest] Error 1
```

但是openssl没有问题，应该是没有定义OPENSSL_NO_MD2，OPENSSL_NO_MD4的问题，如果想兼容openssl，最好继承原有的编译宏。

openldap出问题的代码片段

```
#if OPENSSL_VERSION_NUMBER >= 0x10100000
        md = EVP_get_digestbynid( X509_get_signature_nid( cert ));
#else
        md = EVP_get_digestbynid(OBJ_obj2nid( cert->sig_alg->algorithm ));
#endif

        /* See RFC 5929 */
        if ( md == NULL ||
             md == EVP_md_null() ||
#ifndef OPENSSL_NO_MD2
             md == EVP_md2() ||
#endif
#ifndef OPENSSL_NO_MD4
             md == EVP_md4() ||
#endif
#ifndef OPENSSL_NO_MD5
             md == EVP_md5() ||
#endif
             md == EVP_sha1() )
                md = EVP_sha256();
```

--- a/Configure
+++ b/Configure
@@ -588,8 +588,23 @@ our %disabled = ( # "what"         => "comment"
                   "zkp-gadget"          => "default",
                   "zkp-transcript"      => "default",
                   "bn-method"           => "default",
-                  "smtc"                 => "default",
-                  "smtc-debug"           => "default",
+                  "smtc"                => "default",
+                  "smtc-debug"          => "default",
+                  "aria"                => "default", # ARIA has been removed
+                  "bf"                  => "default", # Blowfish has been removed
+                  "blake2"              => "default", # Blake2 has been removed
+                  "camellia"            => "default", # Camellia has been removed
+                  "cast"                => "default", # CAST has been removed
+                  "idea"                => "default", # IDEA has been removed
+                  "md2"                 => "default", # MD2 has been removed
+                  "mdc2"                => "default", # MDC2 has been removed
+                  "md4"                 => "default", # MD4 has been removed
+                  "gost"                => "default", # GOST has been removed
+                  "rc2"                 => "default", # RC2 has been removed
+                  "rmd160"              => "default", # RIPEMD has been removed
+                  "ripemd"              => "default", # RIPEMD has been removed
+                  "seed"                => "default", # SEED has been removed
+                  "whirlpool"           => "default", # Whirlpool has been removed
                 );
 
 # Note: => pair form used for aesthetics, not to truly make a hash table

